### PR TITLE
feat: createdBy and updatedBy in resource & revision

### PR DESCRIPTION
**Pull request by ishanarya0**
Add createdBy and updatedBy to resource
Add createdBy to revision

### CODE REVIEW

**Comment by ishanarya0** [comment]
[WIP] Updating the [latest proto changes](https://github.com/goto/proton/commit/ec066344b8597f9238dbbfe3cd05532a49df59ca)

**Comment by mabdh** [comment]
@ishanarya0 can you please link the proton changes here?

File: internal/server/serverutils/context.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"context"
+	"net/http"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+const userIDHeader = "user-id"
```
Comment (mabdh) [inline internal/server/serverutils/context.go:13]: should we make this configurable?

**Comment by ishanarya0** [comment]
> @ishanarya0 can you please link the proton changes here?

Proton changes - https://github.com/goto/proton/pull/40

File: internal/server/serverutils/context.go (hunk 1 above)
Comment (ishanarya0) [reply internal/server/serverutils/context.go:13]: sure, we'll make it configurable and release it as a patch later. Will create an issue for this.

**Pull request merged**